FWCore/Framework/src/EventSetupRecordProvider.cc

```diff
@@ -123,7 +103,13 @@ namespace edm {
             make_shared_noexcept_false<IntersectingIOVRecordIntervalFinder>(key_);
         intFinder->swapFinders(*multipleFinders_);
         finder_ = intFinder;
+        hasNonconcurrentFinder_ = intFinder->hasNonconcurrentFinder();
```

Comment:
Wouldn't a IntersectingIOVRecordIntervalFinder which had a finder which was `false == finder->concurrentFinder()` then just make the IntersectingIOVRecordIntervalFinder also do `false == concurrentFinder()`? If so, then why the separate `hasNonconcurrentFinder` ?

Reply:
Thanks. That simplified the code a little. Done.